File: 2020/06/solution.rb
require_relative '../../lib/advent_helper'
helper = AdventHelper.new(script_root:__dir__, script_file: __FILE__)
input = helper.send(:open_file, 'input.txt').read
responses = input.split("\n\n").map { |r| r.split(/\s+/).map(&:chars) }



# Part 1
# 6310
a = responses.sum { |r| r.inject(&:|).count }

# Part 2
# 3193
b = responses.sum { |r| r.inject(&:&).count }



helper.output(a, b)

File: 2020/06/solution.exs
IO.write("\e[H\e[2J")
IO.puts("Running #{__ENV__.file}...")

groups = __DIR__ <> "/" <> "input.txt"
  |> File.read!()
  |> String.split("\n\n", trim: true)
  |> Enum.map(fn g ->
    Enum.map(String.split(g, "\n"), fn p ->
      Enum.uniq(String.graphemes(p))
    end)
  end)

Enum.reduce(groups, 0, fn g, acc ->
  count = Enum.reduce(g, [], fn p, acc ->
    Enum.uniq(acc ++ p)
  end) |> Enum.count
  acc + count
end) |> IO.inspect

Enum.reduce(groups, 0, fn g, acc ->
  count = Enum.reduce(g, fn p, p_acc ->
    Enum.filter(p, fn c -> c in p_acc end)
  end) |> Enum.count
  acc + count
end) |> IO.inspect
